### PR TITLE
Pbi 274 opacity slider

### PR DESCRIPTION
**Pull request by edlu77**

### CODE REVIEW

File: projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+    this.opacityChange.emit(value);
+  }
+
+  /**
+   * Resets all opacity values to 100;
+   */
+  resetOpacity(): void {
+    for(const item of this.items) {
+      item.opacity = 100;
```
Comment (bherr2) [inline projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts:154]: Change detection won't pick this up. Need to make sure you are using best practices to ensure components listening for changes actually see them (it won't be notified when you just update a property of an object, need to set this.items to a new array and not mutating state most likely).

Comment (edlu77) [reply projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts:154]: So basically this?

`
let newItems = this.items;
for(const item of this.newItems) {
      item.opacity = 100;
}
this.items = newItems;
`

Comment (bherr2) [reply projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts:155]: this.items = this.items.map(i => ({ ...i, opacity: 100}));

File: projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div class="item-entry" [class.selected]="selection === item" *ngFor="let item of items" (mouseover)="mouseOver(item)" (mouseout)="mouseOut(item)">
```
Comment (bherr2) [inline projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.html:1]: `*ngFor="let item of items; trackBy: getId"` and add a getId funciton on the class.

**Comment by bherr2** [comment]
Superseded by #337

**Pull request closed without merging**